### PR TITLE
Adds DecompositionPlots to the pymc_marketing.mmm.plotting namespace

### PR DESCRIPTION
**Pull request by isofer**
## Description

Adds `DecompositionPlots` to the `pymc_marketing.mmm.plotting` namespace. This class
exposes three methods for visualising how a fitted MMM model decomposes its target
variable into contributing components.

**Context:** This PR builds directly on `isofer/time-series-plotting`, which introduced
`DiagnosticsPlots` and `TransformationPlots` together with the shared `_helpers.py`
infrastructure.

see [demo notebook here](https://gist.github.com/isofer/ffd1c8ebee47035f74a0c89e561dd12c)

| File | Change |
|---|---|
| `pymc_marketing/mmm/plotting/decomposition.py` | New — 460 lines, `DecompositionPlots` class |
| `tests/mmm/plotting/test_decomposition.py` | New — 380 lines, full test coverage |
| `pymc_marketing/mmm/plotting/__init__.py` | Adds `DecompositionPlots` to `__all__` |

### Methods added

**`contributions_over_time`** — Time-series plot with HDI bands, one panel per
extra-dimension combination (e.g. per geo). Supports filtering to any subset of
`{"channels", "baseline", "controls", "seasonality"}`. Channels are plotted individually;
baseline is broadcast over the date axis; controls are summed to a single time-series.
Returns `(Figure, NDArray[Axes])` or `PlotCollection`.

**`waterfall`** — Horizontal waterfall chart showing mean contribution per component,
one panel per extra-dimension combination. Each bar shows a named component (one bar
per channel, one for baseline, controls, seasonality) and a final "total" bar.
Contributions are drawn in green/red depending on sign; totals are grey. Percentage
labels are placed inside each bar.

**`channel_share_hdi`** — Forest plot of each channel's share of total channel
contribution summed over dates, with 50% and 94% HDI intervals (or user-supplied
`hdi_prob`). Uses `azp.plot_forest` under the hood.

## Related Issue

-  Related to #2458 

## Changes Made

- Add `DecompositionPlots` class with `contributions_over_time`, `waterfall`, and
  `channel_share_hdi` methods.
- Export `DecompositionPlots` from `pymc_marketing.mmm.plotting`.
- Add `tests/mmm/plotting/test_decomposition.py` covering all three methods including
  no-extra-dims and multi-geo fixtures, edge cases (unknown `include` key, `bar_kwargs`
  conflict, empty contribution filter), and `return_as_pc` / `idata` override paths.

----
📚 Documentation preview 📚: https://pymc-marketing--2488.org.readthedocs.build/en/2488/

### CODE REVIEW

**Comment by isofer** [comment]
@cursor push 2dd55f74ea

File: pymc_marketing/mmm/plotting/decomposition.py [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+        for bar_idx, (label, val) in enumerate(components):
+            if label == "total":
+                color = "grey"
+                left = 0.0
+                width = val
+            else:
+                color = "green" if val >= 0 else "red"
+                left = running
+                width = val
+                running += val
```
[the remark this answers is not in the document]
Comment (isofer) [reply pymc_marketing/mmm/plotting/decomposition.py:87]: unrealistic edge case

**Comment by isofer** [comment]
@juanitorduz ready to be reviewed after #2488 (but they are independent).

**Comment by isofer** [comment]
@juanitorduz this is ready to be reviewed

File: pymc_marketing/mmm/plotting/decomposition.py [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+                        (str(val), da.sel({coord_dim: val}).mean(dim=mean_dims))
+                    )
+            else:
+                entries.append((ds_key, da.mean(dim=mean_dims)))
+
+        # Determine panel combos from extra dims
+        if extra_dims:
+            ref_da = entries[0][1]
+            coord_values = [ref_da.coords[d].values for d in extra_dims]
+            combos = list(itertools.product(*coord_values))
```
[the remark this answers is not in the document]
Comment (isofer) [reply pymc_marketing/mmm/plotting/decomposition.py:354]: @cursor push bb1b7ae63b

**Review by juanitorduz — COMMENTED** [review]
LGTM! Thanks @isofer ! Still, this is hard to review as I do not see the plots. Can you please add this new API to the example notebook https://www.pymc-marketing.io/en/latest/notebooks/mmm/mmm_example.html to see if we get the same results?

**Review by juanitorduz — APPROVED** [review]
Actually, I just saw the demo notebook (still, you can try adding this new api inot the example notebook now or in a following PR :) )

**Comment by isofer** [comment]
@cursor push 0c4ab43860

**Comment by isofer** [comment]
@cursor push 4b08652006
